android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/SessionPrioritizationMiddleware.kt

```diff
@@ -16,13 +22,19 @@ import mozilla.components.concept.engine.EngineSession.SessionPriority.HIGH
 import mozilla.components.lib.state.Middleware
 import mozilla.components.lib.state.MiddlewareContext
 import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.base.coroutines.Dispatchers as MozillaDispatchers
 
 /**
  * [Middleware] implementation responsible for updating the priority of the selected [EngineSession]
  * to [HIGH] and the rest to [DEFAULT].
  */
-class SessionPrioritizationMiddleware : Middleware<BrowserState, BrowserAction> {
+class SessionPrioritizationMiddleware(
```

Comment:
It will be nice to have a way to test this in nightly, without the pressure of having to revert in case something goes wrong or we don't want it ride the trains until collecting enough info. Are we going to duplicate this middleware or do we have another way to test the new behaviour and keeping the old one as fallback ?

Reply:
If we create a new middleware with the new implementation, how would we distinguish between using the new vs the old? A flag somewhere?

Reply:
A flag in the constructor could be an alternative too, @jonalmeida do you have any ideas :) ?

Reply:
We can do this in Fenix with a feature flag around the middleware in the list.

Reply:
Should we go then with the flag in the constructor? :)

Reply:
Sorry, when I sent my comment, I hadn't seen any of the other previous messages. Adding a flag in the constructor seems overly complicated to do since we're modifying the existing middleware and a flag around the middleware also doesn't make sense since we need the existing behaviour to continue (I confused this feature with the `trimMemoryAutomatically` flag).

@calumozilla has a good point about distinguishing the old with the new behaviour. Maybe the best way forward is to land this after the next release and monitor crash rates + ask users that had the bug with form data disappearing to reproduce again. We will have time to back out if needed then.

What do you both think?